test_programs/compile_success_empty/infer_int_loop_bounds/src/main.nr

```diff
@@ -0,0 +1,18 @@
+// issue #4290
+fn loop_from<N, K>(_arr: [Field; N], _another: [Field; K]) -> u32 {
+    let mut acc = 0;
+    // N and K are type Field but the iterator is u32
```

Comment:
Can you update this to "were of" type Field? I'd expect they should be inferred to be `u64` now.

Reply:
I think I'm missing something because I don't see how these examples were fixed when this PR doesn't seem to include any unifications to the new `TypeVariableKind::Integer`.

Reply:
Just tested this on master and the test works there as well

Reply:
also tested on master and was able to replicate it working: should I leave this as a regression test then or remove?

Reply:
I'd say remove it and lets add tests for this change when the unification is added later